api/helpers/gamestate/moves/resolve/execute.js

```diff
@@ -59,6 +59,19 @@ module.exports = {
       case 1:
         result = sails.helpers.gamestate.moves.resolve.ace(result);
         break;
+      case 3:
+      case 4:
+      case 5:
+      case 7:
+        // These one-offs require an additional user input after resolving
+        // Each one has its own phase designated by the rank
+        result = {
+          ...result,
+          phase: oneOff.rank,
```

Comment:
This is a bit of a funky move because all the other references to a `GameState`'s `phase` are assigned using the `GamePhase` enum. The trick here is that the values for `GamePhase.RESOLVING_THREE`, `GamePhase.RESOLVING_FOUR`, `GamePhase.RESOLVING_FIVE`, and `GamePhase.RESOLVING_SEVEN` are all equal to their respective one-off rank: 3, 4, 5, and 7. I considered at first making a reverse mapping so this would instead be something like:

```
      phase: rankToGamePhaseMap[oneOff.rank],
```

With the idea being that it would convert the ranks to their respective `GamePhaseValues`, but this would just be an identity mapping each rank to itself. So it seems simpler to use the rank _as_ the `GamePhase`. 

Another alternative would be to separate it into different cases for each of the ranks 3-5 like

```
    case 3:
      result = {
        ...result,
        phase: GamePhase.RESOLVING_THREE,
        oneOff,
        turn: result.turn - 1,
      }
```

But then everything except the phase is assigned exactly the same way for all four cases. This is significantly DRYer so it seems like the best I've come up with, even though it is the only place in the app to reference `phase` without using the `GamePhase` enum

Reply:
I agree that this is DRYer and better